### DISCUSSION TITLE
Unable to generate the PDF error when using Kroki

### DISCUSSION THREAD

**Discussion by woprisalive** [Mogztter/asciidoctor-web-pdf#600] [Q&A]
Hi,

I've installed your amazing asciidoctor-web-pdf on a Fedora Linux 36 (Server Edition Prerelease) with node v16.13.2.

I use node only for this tool. Installation was done with the following command :
`npm i -g @asciidoctor/core asciidoctor-pdf asciidoctor-kroki --save-dev`

When I want to generate a PDF, the error `Unable to generate the PDF - Error: Error: Navigation failed because browser has disconnected!` is returned.

I've got the error with the `piracy.adoc` example on the diagram section. If I comment the nomnomol diagram the PDF is generated.

This error seems to be generated by pupeteer related to a memory error... 

When I look at the top command, several chrome processes are executed that use 16.6g of virtual memory !!

![adoc-top](https://user-images.githubusercontent.com/44974128/150314205-73b0e52e-46d9-4546-98fd-1d4924ab698f.png)

New with node and asciidoctor-web-pdf, I don't know where to look for...
I'm not able to use the `--preview` option, I use the Fedora remotely and cannot export the display.

Bye bye !

**Comment by ggrossetie** [comment]
Hello @woprisalive 

I'm not really sure what is going on. As you've mentioned, the headless Chrome instance is probably exceeding the memory limit of your system and as a result Asciidoctor Web PDF cannot communicate with it (through Puppeteer).

> When I look at the top command, several chrome processes are executed that use 16.6g of virtual memory !!

Probably because Puppeteer wasn't able to properly close the browser instance.
If I'm reading your screenshot correctly, you have more or less 2Gb of memory available on your system? You can manually kill the chrome instances, try to free up some memory and run it again.

**Comment by woprisalive** [comment]
All chrome's instances disapear when Asciidoctor Web PDF exits. They are visible with `top` less than a second or two...

**Comment by woprisalive** [comment]
I've installed 3 distributions on a virtualbox :

- Fedora 35 Server : same error generated
- Fedora 35 Workstation : after installation of .so dependencies, same error
- Debian 11 Server : after installation of .so dependencies, PDF generated with success 🥇 but node.js is in version v12.22.5, on Fedora it is a v16.

Have you heard about some issues on fedora distros or node v16 ?

**Reply by ggrossetie** [reply]
Thanks for the follow-up.
No I'm not aware of any issues with Node 16 but currently the CI is using Node 12 and 14. Since Node 16 is the current LTS, I will create an issue to update the CI to Node 16.

https://github.com/Mogztter/asciidoctor-web-pdf/issues/602

**Comment by woprisalive** [comment]
Hi @Mogztter,

I've tried my last penny on installing node v12.22.9 with nvm on my Fedora Server 36.

Same error :,(

Maybe something wrong on Fedora distors, but cannot say what...

**Reply by ggrossetie** [reply]
You can take a look at: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#chrome-headless-doesnt-launch-on-unix

**Comment by woprisalive** [comment]
Hi @Mogztter,

I've seen this doc but without any success...
I'll move to Debian...

thanks a lot.
